src/main/java/org/opentripplanner/api/resource/GraphEnvelope.java

```diff
@@ -0,0 +1,94 @@
+/* This program is free software: you can redistribute it and/or
+ modify it under the terms of the GNU Lesser General Public License
+ as published by the Free Software Foundation, either version 3 of
+ the License, or (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>. */
+
+package org.opentripplanner.api.resource;
+
+import com.vividsolutions.jts.geom.Coordinate;
+import org.opentripplanner.routing.graph.Graph;
+import org.opentripplanner.routing.graph.Vertex;
+import org.opentripplanner.util.WorldEnvelope;
+import java.io.Serializable;
+
+/**
+ * GraphMetada is first created after OSM is build.
+ *
+ * It has two envelopes. One with all of OSM vertices. This is used in {@link org.opentripplanner.graph_builder.module.StreetLinkerModule} and
+ * {@link org.opentripplanner.graph_builder.module.TransitToTaggedStopsModule} to skip stops which are outside OSM data area.
+ *
+ * During GTFS reading second envelope is expanded to include each TransitStop with {@link #expandToInclude(double, double)}.
+ * TransitStop modes are added to transitModes variables with help of
+ */
+public class GraphEnvelope implements Serializable {
+
+    /**
+     * This is envelope with all of OSM data and after {@link #updateEnvelope()} is called with all of data.
+     */
+    private WorldEnvelope envelope;
+
+    /**
+     * This envelope starts with from envelope but gets expanded with {@link #expandToInclude(double, double)} for each {@link org.opentripplanner.routing.vertextype.TransitStop}.
+     *
+     * In {@link #updateEnvelope()} replaces envelope.
+     */
+    private transient WorldEnvelope newEnvelope;
```

Comment:
We should probably call these fields osmEnvelope and osmAndTransitEnvelope for clarity.